src/main/java/com/suse/saltstack/netapi/datatypes/SaltVersion.java

```diff
@@ -0,0 +1,145 @@
+package com.suse.saltstack.netapi.datatypes;
+
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Represents a salt version
+ */
+public class SaltVersion implements Comparable<SaltVersion> {
+
+    private static Pattern SALT_VERSION_REGEX =
+            Pattern.compile("^(\\d+)\\.(\\d+)\\.(\\d+)([Rr][Cc](\\d+))?$");
+
+    private final int year;
+    private final int month;
+    private final int bugfix;
+    private final Optional<Integer> releaseCandidate;
+
+    /**
+     * Parses a salt version string
+     * @param versionString the salt version string
+     * @return SaltVersion if the versionString is valid or empty Optional if not.
+     */
+    public static Optional<SaltVersion> parse(String versionString) {
```

Comment:
I'm sorry, but now I remember what bothered me about this the most. The Optional<> here is completely unnecessary. It doesn't add anything except complicates the API. Returns null on invalid and return SaltVersion on valid. Drop Optional<> wrapper. If someone wants to wrap this in Optional for some reason, they can do this via `Optional.ofNullable(SaltVersion.parse(str));` in their code.